packages/react-combobox/src/components/Option/useOption.tsx

```diff
@@ -30,25 +30,32 @@ function getValueString(value: string | undefined, children: React.ReactNode) {
  * @param ref - reference to root HTMLElement of Option
  */
 export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElement>): OptionState => {
-  const { id, fluentKey: key, disabled, value } = props;
+  const { disabled, value } = props;
+  const optionRef = React.useRef<HTMLElement>(null);
 
   // context values
-  const idBase = useContextSelector(ListboxContext, ctx => ctx.idBase);
   const multiselect = useContextSelector(ListboxContext, ctx => ctx.multiselect);
   const onOptionClick = useContextSelector(ListboxContext, ctx => ctx.onOptionClick);
   const registerOption = useContextSelector(ListboxContext, ctx => ctx.registerOption);
   const selectedOptions = useContextSelector(ListboxContext, ctx => ctx.selectedOptions);
 
-  // use the id if provided, otherwise construct id from key & idBase
-  const optionId = id || key ? `${idBase}-${key}` : '';
+  // use the id if provided, otherwise use a generated id
+  const defaultId = useId('fluent-option');
+  const id = React.useMemo(() => {
+    return props.id || defaultId;
+
+    // we only want the id to change if props.id changes
+    // otherwise it needs to be consistent across renders for screen readers
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.id]);
```

Comment:
nit: I would simply add `defaultId` to dependencies to avoid ESLint violation. `defaultId` should not change ever anyway.

```ts
  const id = React.useMemo(() => {
    // we only want the id to change if props.id changes
    // otherwise it needs to be consistent across renders for screen readers
    return props.id || defaultId;
  }, [props.id, defaultId]);
```